src/nvim/api/win_config.c

```diff
@@ -280,8 +280,12 @@ Window nvim_open_win(Buffer buffer, Boolean enter, Dict(win_config) *config, Err
       }
     }
   } else {
-    if (!check_split_disallowed_err(curwin, err)) {
-      goto cleanup;  // error already set
+    // Unlike check_split_disallowed_err, ignore `split_disallowed`, as opening a float shouldn't
+    // mess with the frame structure. Still check `b_locked_split` to avoid opening more windows
+    // into a closing buffer, though.
+    if (curwin->w_buffer->b_locked_split) {  // Can't instead check `buf` in case win_set_buf fails!
+      api_set_error(err, kErrorTypeException, "Cannot open a float when closing the buffer");
```

Comment:
I think it makes sense for the API to include error codes where possible. E.g. "E1159" in this case. We haven't done so mostly because choosing/validating new error codes is cumbersome (I have been meaning to merge a script to help with that).

Reply:
Oh I thought it would just be hardcoded. Up to you :)